### PR TITLE
Changing name of WhiteSource to Mend

### PR DESCRIPTION
**Pull request by grendel513**
Updating name if WhiteSource to Mend - this does not touch documentation as there is another pull request for that specifically already.

### CODE REVIEW

**Comment by Maffooch** [comment]
Also would be good to add a breaking change section to the upgrade notes for folks that could be using Whitesource in an API based import. [Here is an example from when the Nessus -> Tenable conversion was made](https://documentation.defectdojo.com/getting_started/upgrading/2.23/)

**Comment by grendel513** [comment]
Thanks @Maffooch - Migration has been added

**Comment by grendel513** [comment]
Thanks @Maffooch - documentation added in upgrade notes

**Comment by grendel513** [comment]
> * unit test scan files that are not related to whitesource/mend should probably not be updated here

I updated those references from whitesource to mend to better align with new name.  functionality is not affected

File: docs/content/en/getting_started/upgrading/2.31.0.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: "Upgrading to DefectDojo Version 2.31.0"
```
Comment (kiblik) [inline docs/content/en/getting_started/upgrading/2.31.0.md:2]: Please update the existing `2.31.md`. This will create duplicity.

**Review by kiblik — CHANGES_REQUESTED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by Maffooch** [comment]
Failing test is accommodated in #9348 
```
2024-01-31T19:46:48.0088609Z [36muwsgi_1         |[0m ======================================================================
2024-01-31T19:46:48.0089177Z [36muwsgi_1         |[0m FAIL: test_file_existence (unittests.test_parsers.TestParsers.test_file_existence) (parser='mend', category='docs')
2024-01-31T19:46:48.0089469Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2024-01-31T19:46:48.0089818Z [36muwsgi_1         |[0m Traceback (most recent call last):
2024-01-31T19:46:48.0090196Z [36muwsgi_1         |[0m   File "/app/unittests/test_parsers.py", line 27, in test_file_existence
2024-01-31T19:46:48.0090371Z [36muwsgi_1         |[0m     self.assertTrue(
2024-01-31T19:46:48.0091186Z [36muwsgi_1         |[0m AssertionError: False is not true : Documentation file '/app/unittests/../docs/content/en/integrations/parsers/file/mend.md' is missing or using 
```

**Review by Maffooch — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Comment by devopscodeck** [comment]
In my tests the functionality is not working... also for Whitesource naming, Mend's json reports are not acceptable for defectdojo, even the json format.

The versions that i tested: defectdojo  3.31.0 && 2.27.4
Somebody else?

**Comment by kiblik** [comment]
There was some issue with "WhiteSource vs Mend" reported in Slack as well https://owasp.slack.com/archives/C2P5BA8MN/p1707244117982809

**Comment by mtesauro** [comment]
@devopscodeck Can you provide a sanitized Mend json file? Having samples is really the only way for us to test scanner output especially for commercial offerings.

Happy to look at the parser code and add unit tests but we'll need an example first.

**Pull request merged**